### ISSUE TITLE
EGUI text and images doesn't look great in some environments. 

### ISSUE THREAD

**Issue by Nazariglez** [Nazariglez/nae#60]
I'm having some trouble integrating EGUI. It doesn't look quite well in different situations:

Using WebGL2 on a screen with `device pixel ratio = 1` looks like this:
![Screenshot 2021-10-09 at 00 18 33](https://user-images.githubusercontent.com/1929191/136633988-a2c8011f-80d6-4d1a-a00d-554befa56f33.png)

While it should look like this:
<img width="303" alt="Screenshot 2021-10-09 at 00 19 10" src="https://user-images.githubusercontent.com/1929191/136634006-948df5bd-6179-47e2-a16b-57be73001ac7.png">

And using `UserTextures` in any platform (web or desktop) it looks like this: 
<img width="315" alt="Screenshot 2021-10-09 at 00 16 52" src="https://user-images.githubusercontent.com/1929191/136634084-1f822013-c078-4ceb-8939-4cae7e4b7547.png">

While it should look like this:
<img width="278" alt="Screenshot 2021-10-09 at 00 17 12" src="https://user-images.githubusercontent.com/1929191/136634116-37174b50-32b6-4ec5-8f74-fff01f321642.png">

I think that it's related to the blend mode. But the impl is almost `1 = 1` to egui glium. The main differences between both implementations are that Notan only supports f32 VBO so the color is passed as a normalized float and then converted back to bytes on the shader: 
https://github.com/Nazariglez/nae/blob/main/notan/crates/notan_egui/src/extension.rs#L48

And that the DPI is managed internally by Notan's graphic implementation (glow) so the EGUIi implementation know nothing about `pixels per point`. While on the glium implementation the dpi is used to calculate the uniforms and the clip rects. 

Pipeline configuration used: https://github.com/Nazariglez/nae/blob/main/notan/crates/notan_egui/src/extension.rs#L110-L124
EGUI texture configuration used: https://github.com/Nazariglez/nae/blob/main/notan/crates/notan_egui/src/extension.rs#L159-L175

I would like to have EGUI fully working with Notan because I see so much potential in this combination. Any help with this issue will be really welcome.

**Comment by emilk** [comment]
The blurry text is curious. It could perhaps be that you need to offset the UV by half a texel. It also looks like you should use a different font_gamma (`1.0`?) at https://github.com/Nazariglez/nae/blob/main/notan/crates/notan_egui/src/extension.rs#L153

The texture looks very much like you are loading the rust PNG as RGBA without premultiplying the alpha. PNG:s use un-multiplied alpha, but the best texture format is premultiplied alpha, so you need to premultiply before uploading the image data as a texture. You should do something like this: https://github.com/emilk/egui/blob/master/egui_glium/examples/native_texture.rs#L23-L38

**Comment by Nazariglez** [comment]
Fixed! I changes the `gamma` for DPI != 2.0 because I saw that even in a desktop environment if the screen's DPI is 1.0 it looks weird. And for the images, I'm converting the textures to how EGUI wants the pixels when the texture is registered. This is new to me, so I'm not sure if it will be safe to do this for all the images loaded from files that are not going to be used for EGUI, so I decided to just clone and convert the ones that are used from EGUI for now.

I was clueless, Thanks @emilk!
